extensions/functions_geometry.yaml

```diff
@@ -0,0 +1,29 @@
+%YAML 1.2
+---
+scalar_functions:
+  -
+    name: "point"
+    description: >
+      Return a 2D point with the given `x` and `y` coordinate values.
+    impls:
+      - args:
+          - name: x
+            value: fp64
+          - name: y
+            value: fp64
+        return: geometry
```

Comment:
`geometry` as a type would only be valid if it was a core substrait type (e.g `i64`). To reference a user-defined type, the syntax would be `u!geometry`.

Currently though, there's a limitation where types from one extension file can't be referenced from a different extension file: https://github.com/substrait-io/substrait/issues/496

This means that you can't reference the `geometry` type from `extension_types.yaml` (or really any of the types) in this file.

One way to work around this for now would be to define `geometry` and the other types we need directly in this file.

Reply:
Thanks!  Made some updates!